homeassistant/components/unifiprotect/__init__.py

```diff
@@ -62,6 +63,12 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     try:
         nvr_info = await protect.get_nvr()
     except NotAuthorized as err:
+        retry_key = f"{entry.entry_id}_auth"
+        retries = hass.data.setdefault(DOMAIN, {}).get(retry_key, 0)
+        if retries < AUTH_RETRIES:
+            retries += 1
+            hass.data[DOMAIN][retry_key] = retries
+            raise ConfigEntryNotReady from err
```

Comment:
note to self: check to see if we enumerate this

Reply:
We do, but we also do a `isinstance` check for the data object since the discovery data is also put in there.

Reply:
Looks like we do in views but it has guards to make sure the data is `ProtectData`

Reply:
We already store DISCOVERY data this way so we can be reasonably sure its ok